src/net_processing.cpp

```diff
@@ -2018,6 +2018,12 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                         tx.GetHash().ToString(), pfrom->id);
                 instantsend.AcceptLockRequest(txLockRequest);
                 instantsend.Vote(tx.GetHash(), connman);
+            } else {
+                CTxLockRequest request(tx);
+                if (request.IsSimple() && CInstantSend::CanAutoLock()) {
+                    if (!instantsend.ProcessTxLockRequest(request, connman))
```

Comment:
Is there a reason why `AcceptLockRequest` and `Vote` are not called in this case?

Reply:
Hmmm, indeed. This was mentioned in one of my earlier reviews but https://github.com/dashpay/dash/pull/2140#pullrequestreview-138003345 I failed to check if it was actually implemented the way I proposed 🙈 TL;DR: we need to "inject" autolock in all logic branches that handle ISes.